nativelib/src/main/resources/scala-native/gc/commix/Heap.c

```diff
@@ -219,9 +219,14 @@ void Heap_Init(Heap *heap, size_t minHeapSize, size_t maxHeapSize) {
 }
 
 void Heap_Collect(Heap *heap) {
+    MutatorThread *mutatorThread = currentMutatorThread;
+    // GC collect triggered during StopTheWorld in interruptible thread might
+    // lead to deadlock It's fine to interrupt thread if it's done before
+    // allocating memory
+    bool wasInterruptible = MutatorThread_setInterruptible(mutatorThread, true);
 #ifdef SCALANATIVE_MULTITHREADING_ENABLED
     if (!Synchronizer_acquire())
-        return;
+        goto done;
```

Comment:
Maybe just repeating the following with a `return;` is better unless you want a really long `else` or something?
```c
MutatorThread_setInterruptible(mutatorThread, wasInterruptible);
```